src/fs/fs.ts

```diff
@@ -320,6 +322,39 @@ export class FileSystem extends EventEmitter implements FlashDataSource {
     this.fileVersions.set(filename, current === undefined ? 1 : current + 1);
   }
 
+  async getPythonProject(): Promise<PythonProject> {
+    const projectName = await this.storage.projectName();
+    const project: PythonProject = {
+      files: {},
+      projectName,
+    };
+    for (const file of await this.storage.ls()) {
+      const data = await this.storage.read(file);
+      const content = new TextDecoder().decode(data);
+      project.files[file] = content;
+    }
+    return project;
+  }
+
+  async replaceWithMultipleFiles(project: PythonProject): Promise<void> {
+    for (const file of await this.storage.ls()) {
+      await this.remove(file);
+    }
+    for (const key in project.files) {
+      this.write(key, project.files[key], VersionAction.INCREMENT);
+    }
+    this._dirty = false;
+    this.project = {
+      ...this.project,
+      id: generateId(),
+    };
+    if (project.projectName) {
+      await this.storage.setProjectName(project.projectName);
+    }
+    await this.overwriteStorageWithFs();
+    return this.notify();
```

Comment:
Maybe this bit can be in common?

Reply:
[`d3b5ca7`](https://github.com/microbit-foundation/python-editor-next/pull/626/commits/d3b5ca73f3f81cfdde7cd50e1effdd04f019cd73)